### PR TITLE
Cleanup secrets and access keys on reuse

### PR DESCRIPTION
**Pull request by fahlmant**
https://issues.redhat.com/browse/OSD-2875

During reuse, any current access key should be revoked and all associated secrets removed. 

WIP: When should the credentials and secrets be recreated? During the reuse code, when an account is claimed, or should we eliminate at least the two IAM user credentials secrets completely?

### CODE REVIEW

**Comment by rogbas** [comment]
@fahlmant could you please rebase this PR to get the latest automated makefile targets for automation? After that, we can easily test/merge this PR.

**Comment by rogbas** [comment]
@jharrington22 @rogbas please comment

File: pkg/controller/accountclaim/reuse.go [hunk 1]
```diff
@@ -96,6 +100,33 @@ func (r *ReconcileAccountClaim) resetAccountSpecStatus(reqLogger logr.Logger, re
 	return nil
 }
 
+func (r *ReconcileAccountClaim) recreateAccountSecrets(reqLogger logr.Logger, reusedAccount *awsv1alpha1.Account) error {
```
Comment (ArielLima) [inline pkg/controller/accountclaim/reuse.go:103]: Am I missing something? Does this method remove existing secrets and recreates them or simply removes existing secrets? Also is it used anywhere?

Comment (fahlmant) [reply pkg/controller/accountclaim/reuse.go:103]: Left over from some testing. Removing

File: pkg/controller/accountclaim/reuse.go [hunk 2]
```diff
@@ -391,3 +464,19 @@ func matchAccountForReuse(account *awsv1alpha1.Account, accountClaim *awsv1alpha
 	}
 	return false
 }
+
+func deleteAllAccessKeys(reqLogger logr.Logger, client awsclient.Client, userName string) error {
```
Comment (ArielLima) [inline pkg/controller/accountclaim/reuse.go:468]: Just a suggestion, what do you think of naming this function something along the lines of `deleteAllUserAccessKeys`?

File: pkg/controller/accountclaim/reuse.go (hunk 1 above)
Comment (ArielLima) [reply pkg/controller/accountclaim/reuse.go:144]: Will this be used?

Comment (fahlmant) [reply pkg/controller/accountclaim/reuse.go:144]: Forgot to remove it

**Comment by iamkirkbater** [comment]
Looks good to me.  Do we want to squash?

**Comment by fahlmant** [comment]
Yep I'll squash and merge next week

**Comment by iamkirkbater** [comment]
Sounds good.  Once you squash let me know and I'll LGTM it.

File: pkg/controller/accountclaim/reuse.go [hunk 3]
```diff
@@ -185,7 +189,55 @@ func (r *ReconcileAccountClaim) cleanUpAwsAccount(reqLogger logr.Logger, awsClie
...
+			awsErrors <- delError
+			return err
+		}
+
+		accessKeyOutput, err := account.CreateUserAccessKey(awsClient, getUserOutput.User)
+		if err != nil {
+			return err
+		}
+
+		secretName := claim.Spec.AccountLink + "-secret"
```
Comment (jharrington22) [inline pkg/controller/accountclaim/reuse.go:215]: We should use this function here https://github.com/openshift/aws-account-operator/blob/ad3cf6c45c8658ea1f12d63ab7c75fe2dc0390ab/pkg/controller/account/iam.go#L691-L695

Comment (fahlmant) [reply pkg/controller/accountclaim/reuse.go:215]: 👍

Comment (fahlmant) [reply pkg/controller/accountclaim/reuse.go:215]: @jharrington22 This is a private func in the account controller. Should we move this to utils?

Comment (jharrington22) [reply pkg/controller/accountclaim/reuse.go:215]: sure makes sense

File: pkg/controller/accountclaim/reuse.go [hunk 4]
```diff
@@ -22,9 +26,13 @@ const (
 	// AccountReady indicates account creation is ready
 	AccountReady = "Ready"
 	// AccountFailed indicates account reuse has failed
-	AccountFailed = "Failed"
+	AccountFailed      = "Failed"
+	osdManagedAdmin    = "osdManagedAdmin"
+	osdManagedAdminSRE = "osdManagedAdminSRE"
 )
 
+var secretSuffixes = []string{"-secret", "-osdmanagedadminsre-secret", "-sre-cli-credentials", "-sre-console-url"}
```
Comment (jharrington22) [inline pkg/controller/accountclaim/reuse.go:34]: Is this needed anymore?

**Pull request merged**